### ISSUE TITLE
VAAPI Decoding Issues

### ISSUE THREAD

**Issue by Queuecumber** [saiarcot895/chromium-ubuntu-build#22]
At some point I was getting hit with VA-API bug [262](https://github.com/01org/intel-vaapi-driver/issues/262), based on vp9 videos looking awful in chromium as well as mpv. After the patch for that was cherry picked over to the v1.8 branch, I built and installed it and now the videos look fine in mpv. 

However *all* videos have issues in chromium (vp9, h264, etc). Colors look weird, the system hangs when starting playback, things like that. They usually resolve after a min or so of playback. I know you are applying a VA-API patch, any idea if we can find out if its related?

As usual (for the 3rd time or something) theres a good chance this isnt really your issue since youre just packaging

**Comment by saiarcot895** [comment]
[This](https://github.com/saiarcot895/chromium-ubuntu-build/blob/master/debian/patches/enable_vaapi_on_linux_2.diff) is the patch being used. Most of the content is coming from [this](https://chromium-review.googlesource.com/c/chromium/src/+/532294) pull request. I do make some changes to the patch, mainly to keep it updated for the latest versions of chromium, but that shouldn't affect _how_ decoding happens; instead, if it changes anything, it'll be whether hardware decoding is used or not.

**Comment by Queuecumber** [comment]
Understood

Any idea what might be going on though? Chromium is the only application that has this issue (decoding colors wrong, hanging the entire system so that it needs to be forced off...) I'm also in touch wit the vaapi guys from intel and they seem stumped too

**Comment by mikhailnov** [comment]
I had similr issues on ubuntu 17.10, some H264 videos decoded well, some were just black. It seems that after upgrading from Chromium 63 to 65 those issues disappeared or I just have not met them yet.

$ vainfo
libva info: VA-API version 0.40.0
libva info: va_getDriverName() returns 0
libva info: Trying to open /usr/lib/x86_64-linux-gnu/dri/i965_drv_video.so
libva info: Found init function __vaDriverInit_0_40
libva info: va_openDriver() returns 0
vainfo: VA-API version: 0.40 (libva )
vainfo: Driver version: Intel i965 driver for Intel(R) CherryView - 1.8.3
vainfo: Supported profile and entrypoints
      VAProfileMPEG2Simple            : VAEntrypointVLD
      VAProfileMPEG2Simple            : VAEntrypointEncSlice
      VAProfileMPEG2Main              : VAEntrypointVLD
      VAProfileMPEG2Main              : VAEntrypointEncSlice
      VAProfileH264ConstrainedBaseline: VAEntrypointVLD
      VAProfileH264ConstrainedBaseline: VAEntrypointEncSlice
      VAProfileH264Main               : VAEntrypointVLD
      VAProfileH264Main               : VAEntrypointEncSlice
      VAProfileH264High               : VAEntrypointVLD
      VAProfileH264High               : VAEntrypointEncSlice
      VAProfileH264MultiviewHigh      : VAEntrypointVLD
      VAProfileH264MultiviewHigh      : VAEntrypointEncSlice
      VAProfileH264StereoHigh         : VAEntrypointVLD
      VAProfileH264StereoHigh         : VAEntrypointEncSlice
      VAProfileVC1Simple              : VAEntrypointVLD
      VAProfileVC1Main                : VAEntrypointVLD
      VAProfileVC1Advanced            : VAEntrypointVLD
      VAProfileNone                   : VAEntrypointVideoProc
      VAProfileJPEGBaseline           : VAEntrypointVLD
      VAProfileJPEGBaseline           : VAEntrypointEncPicture
      VAProfileVP8Version0_3          : VAEntrypointVLD
      VAProfileVP8Version0_3          : VAEntrypointEncSlice
      VAProfileHEVCMain               : VAEntrypointVLD

**Comment by Queuecumber** [comment]
@mikhailnov Are you seeing any libva output printed when chromium is started in a terminal? I also tried again to fix this and I havent had any issues with any codec, however I also dont see any of the usual `libva info:` messages in my terminal so I want to confirm that vaapi is even being used. chrome://media-internals does show GpuVideoDecoder for all the videos im trying though.

**Comment by Queuecumber** [comment]
I have gpu for vp9 as well as h264. We might have to find some way to trace the vaapi calls, i have a sneaking suspicion that it's lying about the decoder

**Comment by mikhailnov** [comment]
probablyt we can watch the GPU load https://medium.com/@niklaszantner/check-your-intel-gpu-usage-via-commandline-11196a7ee827

**Comment by Queuecumber** [comment]
Good call, I'll give that a shot today

**Comment by saiarcot895** [comment]
I was able to confirm that on Dev, the GPU decoder is indeed being used with a vp9 video, even though there's no VAAPI info printed to the console.

**Comment by Queuecumber** [comment]
@saiarcot895 Neat, how did you confirm?

**Comment by Queuecumber** [comment]
@mikhailnov I used that tool, `intel_gpu_top` and I confirmed that there is different GPU behavior with hardware acceleration enabled vs disabled

Enabled
----------
![Hardware accelerated decoding ENABLED](https://user-images.githubusercontent.com/5865387/35478107-c9e5485a-03a2-11e8-9ca2-a21cafe4fbf0.png)

Disabled
----------
![Hardware accelerated decoding DISABLED](https://user-images.githubusercontent.com/5865387/35478110-e3d4c7b8-03a2-11e8-959f-131415f87a62.png)

What all those acronyms mean is not easy to find out, it would have been nice if they had something simple lile "GPU load", "Video decoding load", etc like NVidia does.

Anyway, the fact that they are doing different things is enough for me to be convinced its working. Additionally I tried my old standby, `mpv --hwdec=vaapi` and it did *not* print libva messages, which means that probably it's no longer the default to print them or its been disabled (which is a good thing).

Anyway unless anyone has some further objections I think this is resolved
